src/Core/Animation/KeyFramedValueInterpolators.hpp

```diff
@@ -0,0 +1,125 @@
+#ifndef RADIUMENGINE_KEYFRAMEDVALUE_INTERPOLATOR_HPP
+#define RADIUMENGINE_KEYFRAMEDVALUE_INTERPOLATOR_HPP
+
+#include <Core/Animation/KeyFramedValue.hpp>
+#include <Core/Animation/Pose.hpp>
+#include <Core/Animation/PoseOperation.hpp>
+#include <Core/Math/Interpolation.hpp>
+
+namespace Ra {
+namespace Core {
+namespace Animation {
+
+/**
+ * Look for the keyframes around time \p t.
+ * \param[in] keyframes The list of keyframes.
+ * \param[in] t The time to search for.
+ * \param[out] i0 The index for the keyframe preceding t.
+ * \param[out] i1 The index for the keyframe following t.
+ * \param[out] dt Linear parameter, \f$ \in [0;1] \f$, of \p t between times \p t0
+ *             and \p t1 at keyframes i0 and i1.
+ * \note If \p t is lower than the first keyframe's time tf,
+ *       then \p i0 = \p i1 = 0 and \p dt = 0.
+ *       If \p t is higher than the last keyframe's time tl,
+ *       then \p i0 = \p i1 = the index of the last keyframe and \p dt = 0.
+ *       If \p t is an exact match on a keyframe's time tt,
+ *       then \p i0 = \p i1 = the index of the keyframe and \p dt = 0.
+ */
+template <typename T>
+void findRange( const typename KeyFramedValue<T>::KeyFrames& keyframes,
+                const Scalar& t,
+                size_t& i0,
+                size_t& i1,
+                Scalar& dt ) {
```

Comment:
Why ref to indices (can we modify the indices ? Is so, what about the others keyframevalues of the collection ?) and not const ref to the keyframes (or even non const if wa allow to modifyt some par of the keys?

Reply:
indices are out parameters, so is dt, since the goal of this method is to find them ;)

Reply:
so, at least, const ref ...

Reply:
why would you want const since we want to compute them?
using tuple is overdoing it